currency/storage.go

```diff
@@ -74,14 +73,21 @@ func (s *Storage) SetDefaults() {
 		log.Errorf(log.Currency, "Currency Storage: Setting default cryptocurrencies error: %s", err)
 	}
 	s.SetupConversionRates()
-	s.fiatExchangeMarkets = forexprovider.NewDefaultFXProvider()
+	s.fiatExchangeMarkets = nil
+}
+
+// ForexEnabled returns whether the currency system has any available forex providers enabled
+func ForexEnabled() bool {
+	return storage.fiatExchangeMarkets != nil
 }
 
 // RunUpdater runs the foreign exchange updater service. This will set up a JSON
 // dump file and keep foreign exchange rates updated as fast as possible without
 // triggering rate limiters, it will also run a full cryptocurrency check
 // through coin market cap and expose analytics for exchange services
 func (s *Storage) RunUpdater(overrides BotOverrides, settings *Config, filePath string) error {
+	s.fiatExchangeMarkets = nil
```

Comment:
Yes, essentially, the reason for it is because of tests *and* because RunUpdater is _currently_ only run once, but I don't want to introduce a beartrap for someone later who might have reason to call it twice

Reply:
Seeing what you mean about the lock, I agree. But I also don't want to have all those return errors ignore an already set and now should-be-invaild provider.

As an experiment, how about this: 71a7eedf

The overlap of lock disenganging and reengaging going into ForeignExchangeUpdater remedies my discomfort about that tightly coupled lock dependency anyway.

Reply:
Yeah, okay. That won't fly.
Manifestly Raceable with the lock getting stolen before the goro runs.

I guess this, then: 830b301

It has the downside of skipping the errors, but I guess that's the lesser of 2 wevils 🐁

Reply:
🤷  Yikes. Must have gone wrong in a merge because 830b301 clearly shows it removed 🤷 

Fixed (again) 6eab25c